url.go

```diff
@@ -41,17 +47,19 @@ type URL struct {
 	Series   string // "precise" or "" if unset; "bundle" if it's a bundle.
 }
 
+const bundleSeries = "bundle"
+
 var ErrUnresolvedUrl error = fmt.Errorf("charm or bundle url series is not resolved")
 
 var (
-	validSeries = regexp.MustCompile("^[a-z]+([a-z0-9]+)?$")
+	validSeries = set.NewStrings(series.SupportedSeries()...).Union(set.NewStrings(bundleSeries))
```

Comment:
I see what you mean. Since there's not a backwards-compatibility issue we can do this now and if we come up with a better option before next April we can add that.